server/jetstream_cluster.go

```diff
@@ -4436,7 +4436,11 @@ func (js *jetStream) processClusterCreateConsumer(ca *consumerAssignment, state
 		if isConfigUpdate = !reflect.DeepEqual(&cfg, ca.Config); isConfigUpdate {
 			// Call into update, ignore consumer exists error here since this means an old deliver subject is bound
 			// which can happen on restart etc.
-			if err := o.updateConfig(ca.Config); err != nil && err != NewJSConsumerNameExistError() {
+			// JS lock needed as this can mutate the consumer assignments and race with updateInactivityThreshold.
+			js.mu.Lock()
```

Comment:
These suppose to be part of this PR?

Reply:
Something about the changing in the locking caused this race condition to hit far more often, to the point that it was taking several attempts of kicking CI.

I've kept it isolated in a second commit though, as that change won't be relevant for a 2.10.x backport.